pyo3-macros-backend/src/pyclass.rs

```diff
@@ -2380,19 +2380,14 @@ impl<'a> PyClassImplsBuilder<'a> {
             quote! {
                 impl #pyo3_path::impl_::pyclass::PyClassWithFreeList for #cls {
                     #[inline]
-                    fn get_free_list(py: #pyo3_path::Python<'_>) -> &'static ::std::sync::Mutex<::std::option::Option<#pyo3_path::impl_::freelist::PyObjectFreeList>> {
-                        use #pyo3_path::sync::OnceExt;
-                        static mut FREELIST: ::std::sync::Mutex<::std::option::Option<#pyo3_path::impl_::freelist::PyObjectFreeList>> = ::std::sync::Mutex::new(::std::option::Option::None);
-                        static ONCE: ::std::sync::Once = ::std::sync::Once::new();
+                    fn get_free_list(py: #pyo3_path::Python<'_>) -> &'static ::std::sync::Mutex<#pyo3_path::impl_::freelist::PyObjectFreeList> {
+                        static mut FREELIST: #pyo3_path::sync::GILOnceCell<::std::sync::Mutex<#pyo3_path::impl_::freelist::PyObjectFreeList>> = #pyo3_path::sync::GILOnceCell::new();
```

Comment:
We should not need `static mut` any more, now we use a mutex & interior mutability.

```suggestion
                        static FREELIST: #pyo3_path::sync::GILOnceCell<::std::sync::Mutex<#pyo3_path::impl_::freelist::PyObjectFreeList>> = #pyo3_path::sync::GILOnceCell::new();
```